Description: Add aspect ratio estimates to resolution drop-down menu
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=549642

Index: gnome-control-center-2.31.90/capplets/display/xrandr-capplet.c
===================================================================
--- gnome-control-center-2.31.90.orig/capplets/display/xrandr-capplet.c	2010-09-01 17:42:46.512617000 +1000
+++ gnome-control-center-2.31.90/capplets/display/xrandr-capplet.c	2010-09-01 17:42:59.972617001 +1000
@@ -636,7 +636,43 @@
 static char *
 make_resolution_string (int width, int height)
 {
-    return g_strdup_printf (_("%d x %d"), width, height);
+    int ratio;
+    const char *aspect = NULL;
+
+    if (width && height) {
+        if (width > height) ratio = width * 10 / height;
+        else ratio = height * 10 / width;
+
+        switch (ratio) {
+        case 13:
+            aspect = "4:3";
+            break;
+        case 16:
+            aspect = "16:10";
+            break;
+        case 17:
+            aspect = "16:9";
+            break;
+        case 12:
+            aspect = "5:4";
+            break;
+        // This catches 1.5625 as well (1600x1024) when maybe it shouldn't.
+        case 15:
+            aspect = "3:2";
+            break;
+        case 18:
+            aspect = "9:5";
+            break;
+        case 10:
+            aspect = "1:1";
+            break;
+        }
+    }
+
+    if (aspect)
+        return g_strdup_printf (_("%d x %d (%s)"), width, height, aspect);
+    else
+        return g_strdup_printf (_("%d x %d"), width, height);
 }
 
 static void
